packages/ui/src/components/MessageInput/helpers.ts

```diff
@@ -0,0 +1,96 @@
+import { EditorBridge } from '@10play/tentap-editor';
+import { createDevLogger, tiptap } from '@tloncorp/shared/dist';
+import {
+  Block,
+  Inline,
+  constructStory,
+  isInline,
+} from '@tloncorp/shared/dist/urbit';
+
+import { Attachment } from '../../contexts';
+
+const logger = createDevLogger('processReference', true);
+
+export async function processReferenceAndUpdateEditor({
+  editor,
+  pastedText,
+  matchRegex,
+  processMatch,
+}: {
+  editor: EditorBridge;
```

Comment:
We don't use tentap on web, so we should probably do something other than passing the editor bridge here

Reply:
Added the ability to pass either web or native version, how's that feel? Could also see an argument for the function accepting/returning full editor content instead